src/Dropdown.tsx

```diff
@@ -265,11 +265,21 @@ export function Dropdown ({
   onSelect,
   onRemove
 }: DropdownProps) {
-  const [cta, ctaSet] = React.useState(
-    get(items.filter(i => i.value === value)[0], 'label') ||
+  // computed on first render and only updated when value/items changes
+  const computedCTA = React.useMemo(() => {
+    return (
+      get(items.filter(i => i.value === value)[0], 'label') ||
       placeholder ||
       'Please select'
-  )
+    )
+  }, [value, items, placeholder])
+
+  const [cta, ctaSet] = React.useState(computedCTA)
+
+  // will only call ctaSet if computedCTA changes
+  React.useEffect(() => {
+    ctaSet(computedCTA)
+  }, [computedCTA])
```

Comment:
Wrapping my head around this part. Just read about `useMemo` to learn about it. Is there a reason we need to have computedCTA in a `useMemo` hook vs listening to the same list of dependencies (value, items, placeholder) in the `useEffect` call?

Reply:
That's a great question. Now that I think about it, we could probably do this another way that's slightly more simple...

My initial thinking was just that we need this value to populate cta state synchronously, and I didn't want to duplicate logic. But you're right, after the initial render it's only used in one location, so we could break this down further.

Will push up a thought and we can discuss that!